utils/format/formatSwapiPeople.js

```diff
@@ -0,0 +1,31 @@
+import snakeToLowerCamelCase from "./snakeToLowerCamelCase";
+
+/**
+ * Format swapi person objects into compact format
+ *
+ * @param {{object[]}} people
+ * @returns {{name: string, hairColor: string[], url: string, starship: string}[]}
+ */
+export default function formatSwapiPeople(people) {
+  if (!Array.isArray(people)) return [];
+
+  return people
+    .filter(
+      (p) =>
+        typeof p === "object" &&
+        !Array.isArray(p) &&
+        !isNaN(p.height) &&
+        parseInt(p.height) >= 167 &&
+        p.gender === "male" &&
+        (p.hair_color?.includes("white") || p.hair_color?.includes("brown")) &&
+        p.starships?.length >= 1
```

Comment:
This feels a bit like overkill, less lines makes this a bit more readable in my opinion:

```suggestion
        p?.height &&
        parseInt(p.height ?? 0) >= 167 &&
        p.gender === "male" &&
        (p.hair_color?.includes("white") || p.hair_color?.includes("brown")) &&
        Boolean(p.starships?.length)
```